Subject: [PATCH] "Minor refactor code"
---
Index: src/main/java/Service/SendMail.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/Service/SendMail.java b/src/main/java/Service/SendMail.java
--- a/src/main/java/Service/SendMail.java	(revision f48b8b831de53edb2978428799b1633af6aec1b3)
+++ b/src/main/java/Service/SendMail.java	(date 1736739998378)
@@ -41,11 +41,11 @@
         message.setText("http://localhost:8080/MyMavenProject/Active?id="+id);
         return message;
     }
-    public  Message sendMail (String email , String otp ) throws MessagingException {
+    public Message sendMail (String email, String otp) throws MessagingException {
         message.setFrom(new InternetAddress(senderEmail));
         message.setRecipients(Message.RecipientType.TO, InternetAddress.parse(email));
-        message.setSubject("Mã opt của bạn là : ");
-        message.setText(otp);
+        message.setSubject("[Shopping Web] Mã OTP của bạn");
+        message.setText("Chúng tôi gửi bạn mã OTP là: " + otp);
         return message;
     }
 }
\ No newline at end of file
Index: src/main/java/controller/web/CreatePassword.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/controller/web/CreatePassword.java b/src/main/java/controller/web/CreatePassword.java
--- a/src/main/java/controller/web/CreatePassword.java	(revision f48b8b831de53edb2978428799b1633af6aec1b3)
+++ b/src/main/java/controller/web/CreatePassword.java	(date 1736738968409)
@@ -40,12 +40,11 @@
             String email = (String) session.getAttribute("email");
             try {
                 da.updatePassword(email,password);
-                req.setAttribute("message","Đổi mật khẩu thành công !");
+                req.setAttribute("message","Your new password has been updated successfully");
                 req.getRequestDispatcher( "/templates/login.jsp").forward(req, resp);
             } catch (SQLException e) {
                 throw new RuntimeException(e);
             }
-
         }
     }
 }
Index: src/main/webapp/templates/cart.jsp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/webapp/templates/cart.jsp b/src/main/webapp/templates/cart.jsp
--- a/src/main/webapp/templates/cart.jsp	(revision f48b8b831de53edb2978428799b1633af6aec1b3)
+++ b/src/main/webapp/templates/cart.jsp	(date 1736738968757)
@@ -95,7 +95,7 @@
                         <c:if test="${empty cart.items}">
                             <tr>
                                 <td colspan="4" class="text-center">
-                                    <p> <fmt:message key="NotifyCardEmptu"/> <a href="/"><fmt:message key="BuyNow"/></a></p>
+                                    <p> <fmt:message key="NotifyCardEmptu"/> <a href="/MyMavenProject_war/"><fmt:message key="BuyNow"/></a></p>
                                 </td>
                             </tr>
                         </c:if>
Index: src/main/webapp/templates/placeOrder.jsp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/webapp/templates/placeOrder.jsp b/src/main/webapp/templates/placeOrder.jsp
--- a/src/main/webapp/templates/placeOrder.jsp	(revision f48b8b831de53edb2978428799b1633af6aec1b3)
+++ b/src/main/webapp/templates/placeOrder.jsp	(date 1736738968431)
@@ -82,7 +82,7 @@
 <%--                  <input type="text" placeholder="Type here" class="form-control" name="lastName">--%>
 <%--                </div>--%>
 <%--                <div class="form-group col-sm-6">--%>
-<%--                  <label>Phone</label>--%>
+<%--                  <label>Phone Number</label>--%>
 <%--                  <input type="text" value="+998" class="form-control" name="phone">--%>
 <%--                </div>--%>
 <%--                <div class="form-group col-sm-6">--%>
Index: src/main/java/controller/web/ConfirmOTP.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/controller/web/ConfirmOTP.java b/src/main/java/controller/web/ConfirmOTP.java
--- a/src/main/java/controller/web/ConfirmOTP.java	(revision f48b8b831de53edb2978428799b1633af6aec1b3)
+++ b/src/main/java/controller/web/ConfirmOTP.java	(date 1736738968408)
@@ -26,7 +26,7 @@
     protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
         String email = request.getParameter("email");
 
-        request.getRequestDispatcher("/templates/forgotPassword.jsp").forward(request, response);
+        request.getRequestDispatcher("/templates/forgetPassword.jsp").forward(request, response);
     }
 
     @Override
@@ -37,9 +37,8 @@
 
         if (otp.equalsIgnoreCase(sessionOTP)) {
             request.getRequestDispatcher("/templates/createNewPassword.jsp").forward(request, response);
-        }
-        else {
-            request.getRequestDispatcher("/templates/forgot.jsp").forward(request, response);
+        } else {
+            request.getRequestDispatcher("/templates/forgetPassword.jsp").forward(request, response);
         }
     }
 }
\ No newline at end of file
Index: src/main/webapp/templates/createNewPassword.jsp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/webapp/templates/createNewPassword.jsp b/src/main/webapp/templates/createNewPassword.jsp
--- a/src/main/webapp/templates/createNewPassword.jsp	(revision f48b8b831de53edb2978428799b1633af6aec1b3)
+++ b/src/main/webapp/templates/createNewPassword.jsp	(date 1736738968422)
@@ -17,32 +17,32 @@
 <section class="section-conten padding-y" style="min-height: 84vh">
     <div class="card mx-auto" style="max-width: 380px; margin-top: 100px">
         <div class="card-body">
-            <h4 class="card-title mb-4">Tạo mật khẩu mới</h4>
+            <h4 class="card-title mb-4">Create a new password</h4>
             <form action="${pageContext.request.contextPath}/templates/CreatePassword" method="Post">
                 <div class="form-row">
-                    <label for="password">Mật khẩu mới:</label>
+                    <label for="password">New password:</label>
                     <input type="password" name="password" id="password" class="form-control"
                            onchange="validatePassword()" required>
-                    <p id="password-error" style="color: red; display: none;">Mật khẩu phải có từ 8 đến 20 ký tự, bao
-                        gồm chữ hoa, chữ thường, số và ký tự đặc biệt.</p>
+                    <p id="password-error" style="color: red; display: none;">The password must be between 8 and 20 characters, including uppercase letters,
+                        usually, numbers, and special characters.</p>
 
                 </div>
                 <div class="form-row">
-                    <label>Nhập lại mật khẩu mới  </label>
+                    <label>Confirm new password  </label>
                     <input id="Checkpassword" name="Checkpassword" class="form-control" type="password"
                            onChange="validateCheckPassword()" required>
-                    <p id="Checkpassword-error" style="color: red; display: none;"> Mật khẩu không trùng khớp </p>
+                    <p id="Checkpassword-error" style="color: red; display: none;">Passwords do not match</p>
                 </div>
                 <div class="form-group mt-4">
                     <button type="submit" class="btn btn-primary btn-block">
-                        Cập nhật mật khẩu
+                        Update your password
                     </button>
                 </div>
             </form>
         </div>
     </div>
     <p class="text-center mt-4">
-        Don'\'t have account? <a href="register.jsp">Sign up</a>
+        Don't have an account? <a href="register.jsp">Sign up</a>
     </p>
     <br/><br/>
 </section>
Index: src/main/java/controller/web/login.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/controller/web/login.java b/src/main/java/controller/web/login.java
--- a/src/main/java/controller/web/login.java	(revision f48b8b831de53edb2978428799b1633af6aec1b3)
+++ b/src/main/java/controller/web/login.java	(date 1736738969018)
@@ -62,7 +62,7 @@
              }
                 User user = udao.getLogin(email,realPass);
                 if (user == null) {
-                    String message = "Sai thông tin tài khoản mật khẩu ";
+                    String message = "Incorrect email or password";
                     req.setAttribute("message", message);
                     req.getRequestDispatcher("/templates/login.jsp").forward(req,resp);
 
@@ -79,8 +79,8 @@
                     }
                     else {
                         CartDAO cdao = new CartDAO(connection);
-                        Cart cart=  cdao.getCartByUserId(userId);
-                        session.setAttribute("count", cart.getItems().size());
+                        Cart cart = cdao.getCartByUserId(userId);
+                        session.setAttribute("count", cart != null ? cart.getItems().size() : 0);
 
                         resp.sendRedirect(req.getContextPath());
                     }
Index: src/main/webapp/templates/forgetPassword.jsp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/webapp/templates/forgetPassword.jsp b/src/main/webapp/templates/forgetPassword.jsp
--- a/src/main/webapp/templates/forgetPassword.jsp	(revision f48b8b831de53edb2978428799b1633af6aec1b3)
+++ b/src/main/webapp/templates/forgetPassword.jsp	(date 1736738968426)
@@ -9,21 +9,22 @@
     <jsp:include page="headerResource.jsp"/>
 </head>
 <body>
+<jsp:include page="includes/navbar.jsp"/>
 <section class="section-conten padding-y" style="min-height:84vh">
 	<div class="card mx-auto" style="max-width: 380px; margin-top:100px;">
       <div class="card-body">
-      <h4 class="card-title mb-4">Sign in</h4>
+      <h4 class="card-title mb-4">Forget Password</h4>
       <form action="${pageContext.request.contextPath}/templates/Mail" method="post">
           <div class="form-group">
-			 <input type="email" class="form-control" placeholder="Email Address" name="email" >
+			 <input type="email" class="form-control" placeholder="Enter your email" name="email" >
           </div> <!-- form-group// -->
           <div class="form-group">
-              <button type="submit" class="btn btn-primary btn-block"> Send Email  </button>
+              <button type="submit" class="btn btn-primary btn-block"> Get OTP  </button>
           </div> <!-- form-group// -->
       </form>
       </div> <!-- card-body.// -->
     </div> <!-- card .// -->
-     <p class="text-center mt-4">Don'\'t have account? <a href="register.jsp">Sign up</a></p>
+     <p class="text-center mt-4">Don't have an account? <a href="register.jsp">Sign up</a></p>
      <br><br>
 </section>
 <footer class="section-footer border-top padding-y">
Index: src/main/webapp/templates/register.jsp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/webapp/templates/register.jsp b/src/main/webapp/templates/register.jsp
--- a/src/main/webapp/templates/register.jsp	(revision f48b8b831de53edb2978428799b1633af6aec1b3)
+++ b/src/main/webapp/templates/register.jsp	(date 1736738968600)
@@ -23,7 +23,7 @@
           <form action="register" method="POST">
             <div class="form-row">
               <div class="col form-group">
-                <label>First namne </label>
+                <label>First name </label>
                 <input
                   name="FirstName"
                   type="text"
@@ -63,7 +63,7 @@
               <p  style="color: red;">${message}</p>
             </div>
             <div class="form-group">
-              <label>phone</label>
+              <label>Phone Number</label>
              <input name="phone" type="text" id="phone" placeholder="Enter your phone" oninput="validatePhone()"  required   class="form-control" />
              <span id="error-phone" style="color: red; display: none;"></span>
 
Index: src/main/webapp/templates/ConfirmOTP.jsp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/webapp/templates/ConfirmOTP.jsp b/src/main/webapp/templates/ConfirmOTP.jsp
--- a/src/main/webapp/templates/ConfirmOTP.jsp	(revision f48b8b831de53edb2978428799b1633af6aec1b3)
+++ b/src/main/webapp/templates/ConfirmOTP.jsp	(date 1736738968420)
@@ -9,21 +9,22 @@
   <jsp:include page="headerResource.jsp"/>
 </head>
 <body>
+<jsp:include page="includes/navbar.jsp"/>
 <section class="section-conten padding-y" style="min-height:84vh">
   <div class="card mx-auto" style="max-width: 380px; margin-top:100px;">
     <div class="card-body">
       <h4 class="card-title mb-4">Confirm OTP</h4>
       <form action="${pageContext.request.contextPath}/templates/ConfirmOTP" method="post">
         <div class="form-group">
-          <input type="text" class="form-control" placeholder="Email Address" name="otp" >
+          <input type="text" class="form-control" placeholder="Fill your OTP" name="otp" >
         </div> <!-- form-group// -->
         <div class="form-group">
-          <button type="submit" class="btn btn-primary btn-block"> Send Email  </button>
+          <button type="submit" class="btn btn-primary btn-block"> Confirm  </button>
         </div> <!-- form-group// -->
       </form>
     </div> <!-- card-body.// -->
   </div> <!-- card .// -->
-  <p class="text-center mt-4">Don'\'t have account? <a href="register.jsp">Sign up</a></p>
+  <p class="text-center mt-4">Don't have an account? <a href="register.jsp">Sign up</a></p>
   <br><br>
 </section>
 <footer class="section-footer border-top padding-y">
Index: src/main/webapp/templates/login.jsp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/webapp/templates/login.jsp b/src/main/webapp/templates/login.jsp
--- a/src/main/webapp/templates/login.jsp	(revision f48b8b831de53edb2978428799b1633af6aec1b3)
+++ b/src/main/webapp/templates/login.jsp	(date 1736738968428)
@@ -22,7 +22,7 @@
             <input type="password" class="form-control" placeholder="Password" name="password">
           </div>
           <div class="form-group">
-            <a href="forgetPassword.jsp" class="float-right">Forgot password?</a>
+            <a href="forgetPassword.jsp" class="float-right">Forget password?</a>
           </div>
           <div class="form-group">
             <button type="submit" class="btn btn-primary btn-block"> Login </button>
@@ -33,7 +33,7 @@
         </form>
       </div>
     </div>
-    <p class="text-center mt-4">Don'\'t have account? <a href="register.jsp">Sign up</a></p>
+    <p class="text-center mt-4">Don't have an account? <a href="register.jsp">Sign up</a></p>
     <br><br>
   </section>
 <%@ include file="/templates/includes/footer.jsp" %>
